src/data-types/timen.js

```diff
@@ -77,17 +77,22 @@ module.exports = {
     }
   },
 
-  validate: function(value) {
-    if (value == null) {
+  validate(value) {
+    if (value === undefined || value === null) {
       return null;
     }
+
+    let dateValue;
     if (value instanceof Date) {
-      return value;
+      dateValue = value;
+    } else {
+      dateValue = new Date(Date.parse(value));
```

Comment:
Need to remove this redundant conversion to [Date](https://github.com/tediousjs/tedious/pull/677/files#diff-76524e539b02f72f0abdc709930eb753R46) too 😄

Reply:
Yeah. I'm just keeping it because the old code allowed passing in strings as values for `Time`. We can remove this in a breaking change later. 👍

Reply:
I brought that up because [datetime2n](https://github.com/tediousjs/tedious/pull/677/files#diff-1ba56de971ccce1d484f6cb0debb4f72L49) and  [datetimeoffsetn](https://github.com/tediousjs/tedious/pull/677/files#diff-5f9fdee646ff4bda759de09fc175797fL43) had conversion removed. Shouldn't the same behaviour apply for all 3 datatype?

Reply:
Yeah, unfortunately I think that there might be users out there that use `time` and pass in timestrings. I'm not 100% sure, and I'd rather not break this right now, but sometime in the future. 😅

Reply:
😅 I think I should elaborated a bit more.  If users are passing `time` in timestrings format, then it is likely they will do the same for `datetime2` and `datetimeoffset`, as they all have nano second precision.

Code fragment from timen.js and datetimeoffsetn.js looks like,

```node
validate(value) {
  ...
  dateValue = new Date(Date.parse(value));
  ...
}

writeParameterData: function(..){
  ..
  if (parameter.value != null) {
    const time = new Date(+parameter.value); // datetime2n.js doesn't have this conversion
    ...
}
```
We should add the conversion(pointed in the code fragment) back into [datetime2n.js](https://github.com/tediousjs/tedious/pull/677/files#diff-1ba56de971ccce1d484f6cb0debb4f72L49) to allow string passed into `datetime2`.

Reply:
Ah, yes, you're totally right! ❤️ Fixed this in 0d4e197ad58566b9e50c837f61c88aa49a3ffa2a.

Reply:
i think all 3 types now use `Date.parse` in the `validate` method to allow conversion from strings, and they no longer perform redundant conversion inside the `writeParameterData` method.